src/blk/pmem/PMEMDevice.cc

```diff
@@ -168,7 +243,22 @@ bool PMEMDevice::support(const std::string &path)
 {
   int is_pmem = 0;
   size_t map_len = 0;
-  void *addr = pmem_map_file(path.c_str(), 0, PMEM_FILE_EXCL, O_RDONLY, &map_len, &is_pmem);
+  void *addr;
+  int local_fd;
+  int r = 0;
+
+  local_fd = ::open(path.c_str(), O_RDWR);
+  if (local_fd < 0) {
+    return false;
+  }
+
+  r = pmem_check_file_type(local_fd, path.c_str(), NULL);
+  VOID_TEMP_FAILURE_RETRY(::close(local_fd));
+  if (r == 0) {
+    return true;
+  }
```

Comment:
why add those code?

Reply:
checking whether it is a devdax file. then use different open flags